lib/dangermattic/plugins/manifest_pr_checker.rb

```diff
@@ -99,6 +99,23 @@ def check_swift_package_resolved_updated_strict(manifest_path:, manifest_lock_pa
       )
     end
 
+    # Check for duplicate Package.resolved files across all provided SwiftPM package directories
+    #
+    # @param report_type [Symbol] (optional) The type of report for the message. Types: :error, :warning (default), :message.
+    #
+    # @return [void]
+    def check_swift_duplicate_package_resolved(report_type: :warning)
+      package_files = Dir.glob('./**/Package.swift')
+      resolved_files = Dir.glob('./**/Package.resolved')
+      return if package_files.length == resolved_files.length
```

Comment:
Thinking about it more, this is actually still not the right test, because one can have a `Package.resolved` that is generated by `xcodebuild` within the `.xcworkspace` or `.xcodeproj` (if those package dependencies are managed by Xcode), in which case there will be a `Package.resolved` file within the `.xcworkspace/**` folder without a corresponding `Package.swift` in the repo, and the counts won't match.

I see multiple use cases:
 - Project fully managed by SwiftPM, not `xcodeproj` nor `xcworkspace`: we expect `count(Package.swift) == count(Package.resolved)`
 - Project containing only a `xcodeproj`: we expect only a `Package.resolved` within the `xcodeproj/`, and should report a violation if there's a `Package.resolved` outside the `xcodeproj` (e.g. at the root of the repo) without a corresponding `Package.swift` in the same directory
  - Project containing a `xcworkspace` and one or more `xcodeproj`: not sure if we expect only a `Package.resolved` in the `xcworkspace/**` or if there's a possibiilty that Xcode would _also_ generate a `Package.resolved` in the `.xcodeproj`  bundles that the `xcodeproj` references (especially if each `xcodeproj` might have separate Swift packages declared in the "Package Dependencies" tab of each of the `xcodeproj`…?)

So we should cover all the use cases possible (repo with only SwiftPM and no Xcode; repo with a mix of SwiftPM and Xcode project/workspace for the same product; repo with an Xcode project/workspace at the root + one or more `Package.swift` / `Package.resolved` pairs to develop local packages as a way to modularize development of the product; repo with only Xcode project/workspace without any SPM at all; …)

Reply:
> Thinking about it more, this is actually still not the right test, because one can have a Package.resolved that is generated by xcodebuild within the .xcworkspace or .xcodeproj (if those package dependencies are managed by Xcode), in which case there will be a Package.resolved file within the .xcworkspace/** folder without a corresponding Package.swift in the repo, and the counts won't match.

Yeah, I was realizing that earlier today when I looked at the PR again, but didn't get back to it 😕 I'll give it another try.